app/uk/gov/hmrc/transitmovementseisstub/controllers/MessagesController.scala

```diff
@@ -75,29 +80,27 @@ class MessagesController @Inject() (appConfig: AppConfig, cc: ControllerComponen
       routeToStub
   }
 
-  private def routeToStub(implicit request: Request[Source[ByteString, _]]) = {
-    request.body.runWith(Sink.ignore)
-
```

Comment:
We need to be _really_ careful here -- because we need to run the request _once_ otherwise we'll cause a memory leak -- so if we send a message and it's not valid before the LRN is checked, we keep the request around, and eventually the stub would run out of memory.

The error case in the Left case should run this, and then just ignore any errors that occur.

Reply:
This still needs to be done.

Reply:
Sorry @drnaylor-oc am still unclear of the above ask. Please could you explain it bit more.

Reply:
When getting a stream from a request, we **must** use it once. Previously in the stub, we weren't using the XML at all, so we just ran this to ignore it. Now we are using it, in the fully happy path or if the LRN is invalid, the `validateLRN` method will use the stream. However, if one of the previous conditions fail, we never get to `validateLRN`. This means that the stream is never consumed, and Play holds onto it forever, meaning a memory leak occurs.

If any of the previous conditions fail, we need to run the stream with the ignore sink.

Reply:
Updated it, please take a look